### PR TITLE
Fix benchmark reporting

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This Pr fixes benchmark result reporting by adding the step of combining workflow artifacts into a single archive and adding permissions to `benchmark_report.yml` workflow. It cannot be fully tested without merging. Partial merge was present in, #6875

### CODE REVIEW

File: .github/workflows/benchmarks_report.yml [hunk 1]
```diff
@@ -29,6 +33,8 @@ jobs:
                run_id: context.payload.workflow_run.id,
             });
             let artifactName = `asv-benchmark-results-${context.payload.workflow_run.id}-${context.payload.workflow_run.run_number}-${context.payload.workflow_run.run_attempt}`
+            console.log(`Artifact name: ${artifactName}`);
+            console.log(`All artifacts: ${JSON.stringify(allArtifacts.data.artifacts)}`);
             let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
               return artifact.name == artifactName
             })[0];
```
Comment (psobolewskiPhD) [inline .github/workflows/benchmarks_report.yml:40]:
```suggestion
            })[0];
            if (matchArtifact === undefined) {
              throw TypeError('Build Artifact not found!');
            }
```

How about checking the artifact name matches?

**Comment by psobolewskiPhD** [comment]
Maybe add:
```
  push:
    paths:
      - '.github/workflows/benchmarks**'
```
to `benchmarks.yml`?
Then if there is a follow-up it might be easier?

**Review by psobolewskiPhD — APPROVED** [review]
Seems good -- I left a suggestion and question/comment I couldn't suggest.

**Comment by Czaki** [comment]
> to `benchmarks.yml`?
> Then if there is a follow-up it might be easier?

As the trigger is only to add a label and a single run is 3h of CI work, I'm not sure if it is required.

**Review by brisvag — APPROVED** [review]

**Comment by Czaki** [comment]
It works again https://github.com/napari/napari/pull/6727#issuecomment-2085363545

**Pull request merged**